sql/src/main/java/org/apache/druid/sql/calcite/planner/SqlResourceCollectorShuttle.java

```diff
@@ -87,13 +88,21 @@ public SqlNode visit(SqlIdentifier id)
         if (qualifiedNameParts.size() == 2) {
           final String schema = qualifiedNameParts.get(0);
           final String resourceName = qualifiedNameParts.get(1);
+
+          // Put the lookup names in the query context to facilitate selective loading of lookups.
+          if (schema.equals(NamedLookupSchema.NAME)) {
+            plannerContext.queryContextMap().putIfAbsent(PlannerContext.CTX_LOOKUPS_TO_LOAD, new HashSet<>());
+            Set<String> lookupsToLoad = (Set<String>) plannerContext.queryContextMap().get(PlannerContext.CTX_LOOKUPS_TO_LOAD);
+            lookupsToLoad.add(resourceName);
```

Comment:
```suggestion
            plannerContext.queryContextMap().computeIfAbsent(PlannerContext.CTX_LOOKUPS_TO_LOAD, key -> new HashSet<>()).add(resourceName);
```

Reply:
I'm not sure I follow. There can be multiple lookup names which have to be added to the context map. So we have to put value in the context map even if `PlannerContext.CTX_LOOKUPS_TO_LOAD` key is already present.

Please let me know if I'm misunderstanding something. Thanks!

Reply:
The logic here has been updated a bit to move the `putIfAbsent()` part outside of the `if` condition, so this shouldn't be needed now.

Reply:
This is outdated flow now, hence closing this comment.